src/state.ts

```diff
@@ -28,10 +28,29 @@ class State {
         },
     ]
     last_ex_str = "echo"
+    globalMarks: Map<MarkKey, {
+        url: string,
+        scrollX: string,
+        scrollY: string
+        tabId: number
+    }> = new Map()
+    localMarks: Map<MarkUrl,
+        Map<MarkKey, {
+            scrollX: number,
+            scrollY: number
+        }>> = new Map()
+    beforeJumpMark: {
+        scrollX: number,
+        scrollY: number,
+        tabId: number
+    } = undefined
 }
 
+type MarkKey = string;
+type MarkUrl = string;
```

Comment:
I would personally just use `string` directly - do you see any major advantage to making custom types?

Reply:
Removed. My intention was to make it clear what the keys in the maps are,